src/task/image.ts

```diff
@@ -0,0 +1,35 @@
+import { type User } from '../api/auth/authorization.js';
+import { ImageModel } from '../api/db/models/Image.js';
+import { TaskInput } from '../api/db/models/Task.js';
+import type * as gql from '../@types/graphql.js';
+
+export async function DeleteImagesByFilter(task: TaskInput<gql.DeleteImagesByFilterTaskInput>) {
+  const context = { user: { is_superuser: true, curr_project: task.projectId } as User };
+  let images = await ImageModel.queryByFilter(
+    { filters: task.config.filters, limit: 100 },
+    context,
+  );
+  while (images.results.length > 0) {
+    const batch = images.results.map((image) => image._id);
+    await ImageModel.deleteImages({ imageIds: batch }, context);
+    if (images.hasNext) {
+      images = await ImageModel.queryByFilter(
+        { filters: task.config.filters, limit: 100, next: images.next },
+        context,
+      );
+    } else {
+      break;
+    }
+  }
+
+  return { isOk: true };
+}
+
+export async function DeleteImages(task: TaskInput<gql.DeleteImagesInput>) {
+  const context = { user: { is_superuser: true, curr_project: task.projectId } as User };
+  while (task.config.imageIds?.length && task.config.imageIds.length > 0) {
+    const batch = task.config.imageIds?.splice(0, 100);
```

Comment:
I don't know if this matters whatsoever but it feels a little weird to be mutating this input directly with `splice` like that. Maybe we clone the `task.config.imageIds` outside of the while loop and use that instead? Maybe I am just being insane? No strong feelings.

Reply:
Had similar idea, and i committed locally but didnt push haha. Yea we need to preserve the list of image ids to add it to the task output